load-gen/src/main/java/io/smartcat/data/loader/api/RateGenerator.java

```diff
@@ -0,0 +1,6 @@
+package io.smartcat.data.loader.api;
+
+public interface RateGenerator {
+
+    long getRate(long prevInstant, long currentInstant);
```

Comment:
Since x is not same in both cases. In sine, it represents repeating value from 0 to 2pi. And it case of linear, it represents monotonic growing value.